dht_net.go

```diff
@@ -239,15 +267,15 @@ func (dht *IpfsDHT) sendRequest(ctx context.Context, p peer.ID, pmes *pb.Message
 		metrics.SentBytes.M(int64(pmes.Size())),
 		metrics.OutboundRequestLatency.M(float64(time.Since(start))/float64(time.Millisecond)),
 	)
-	dht.peerstore.RecordLatency(p, time.Since(start))
+	m.host.Peerstore().RecordLatency(p, time.Since(start))
```

Comment:
do we know when the 1st byte actually goes out to host? if there are queued messages to that peer on the transport we may end up over-estimating here, right?

Reply:
I don't think so, although I haven't look deeply into the metrics or hooks our transports expose.

However, it's arguable that message queuing is a reasonable thing to include in this latency calculation. For example, if my low throughput to you using Transport X is causing high latency to receiving responses from you it's something I may need to consider in deciding what to send. I'd prefer the more precise option, but not sure how painful that would be.

Either way, something for another PR 😄